### PR TITLE
Convert the inactive_client_timeout test to use Proxy Verifier

### PR DESCRIPTION
**Pull request by bneradt**
Convert the current client inactivity timeout test to use Proxy Verifier
instead of a combination of curl and nc shell scripts.

### CODE REVIEW

**Review by shinrich — DISMISSED** [review]
Looks good.  Much more readable.

**Comment by bneradt** [comment]
[approve ci]

File: tests/gold_tests/timeout/slow_server.yaml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+              {name: tls},
+              {name: tcp},
+              {name: ip} ]
+
+  transactions:
+
+  - client-request:
+      headers:
+        fields:
+        - [ :method, GET ]
```
Comment (masaori335) [inline tests/gold_tests/timeout/slow_server.yaml:160]: `POST` ?

Comment (bneradt) [reply tests/gold_tests/timeout/slow_server.yaml]: Ah yes, copy and paste error. Good catch Masaori.

**Comment by bneradt** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

**Review by masaori335 — APPROVED** [review]
Nice cleanup.

**Review by maskit — COMMENTED** [review]
I'm not going to block this, but 150 lines for these three things? And it requires you to know H2 specific header names? I would not say it's great.
```
self.httpbin = Test.MakeHttpBinServer("httpbin")
curl --http1.1 https://httpbin/delay/10
curl --http2 https://httpbin/delay/10
```

**Comment by bneradt** [comment]
> I'm not going to block this, but 150 lines for these three things? And it requires you to know H2 specific header names? I would not say it's great.
> 
> ```
> self.httpbin = Test.MakeHttpBinServer("httpbin")
> curl --http1.1 https://httpbin/delay/10
> curl --http2 https://httpbin/delay/10
> ```

Thanks for the review.

A quick note to your comment: Proxy Verifier does not require you to know the HTTP pseudo headers. It will produce them from the `method:`, `status:`, etc. nodes if they are used. But in our context it's clearer to explicitly state them for HTTP/2 because we're pretty HTTP/2 literate. That is, for HTTP/2, I'd prefer the replay files to explicitly specify the HTTP pseudo headers rather than them being generated indirectly from other nodes.

That said I'm not opposed to httpbin. It's a good tool to keep around, but I think the version we have available in CI doesn't support delay for the various methods we want to test here. We run with v0.7.0 and the extended delay support is in an unversioned commit on master after that. I hope you agree that this patch expresses the test more naturally than what Susan was constrained to do without Proxy Verifier's (or httpbin's) delay directives.

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**